Tools/plugins/MuonPogTreeProducer.cc

```diff
@@ -369,6 +369,7 @@ void MuonPogTreeProducer::analyze (const edm::Event & ev, const edm::EventSetup
     {
       nGoodMuons = fillMuons(muons,vertexes,beamSpot);
     }
+  eventId_.nMuons = nGoodMuons;
```

Comment:
Hi, ok:
I'm not sure how you are doing this right now, but, if you don't override the default parameter [here](https://github.com/cms-MuonPOG/MuonPOGtreeProducer/blob/80X/Tools/test/muonPogNtuples_cfg.py#L45), nGoodMuons counts the number of muons above 5 GeV only.
Such a threshold is as the one of the TnP producer and "in line" with the typical lower limit where loose muons are usually used (e.g. H=>ZZ).
Typically you might want to cut there (or not too far), but Raffaella needs pT cuts that are significantly higher for her study.
Therefore if she stores in eventId the list of pTs of all the muons in the ntuple, she can quickly loop and check that there are at least 2, let's say, with more than 50 GeV.
The advantage of saving this information into eventId is that she can load just the eventId class from the tree and avoid loading event one which is what dominates the ntuple size (and takes more time to be put in memory).
This should speed-up her analysis workflow when running on ntuples produced for general purposes.
Does this clarify the proposal?